app/controllers/api/v1/podcasts_controller.rb

```diff
@@ -16,12 +17,24 @@ def create
 
     def show
         podcast = Podcast.find(params[:id])
-        render json: podcast
+        if current_user
+            user = serialized_data(current_user, UserSerializer)
+        else
+            user = {user: {id: "", user_name: "", admin: false} }
+        end
+        render json: {
+            podcast: serialized_data(podcast, PodcastSerializer),
+            user: user
+        }
     end
 
     private
 
     def podcast_params
         params.require(:podcast).permit(:name, :url)
     end
+
+    def serialized_data(data, serializer)
+        ActiveModelSerializers::SerializableResource.new(data, each_serializer: serializer)
+      end
```

Comment:
`end` needs to be de-tabbed, but I think Shruthi might have changed this during our quick call / review already